src/Sylius/Behat/Client/RequestFactory.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Client;
+
+use Symfony\Component\HttpFoundation\Request as HttpRequest;
+
+class RequestFactory implements RequestFactoryInterface
+{
+    public function index(
+        ?string $section,
+        string $resource,
+        string $authorizationHeader,
+        ?string $token = null
+    ): RequestInterface {
+        $builder = RequestBuilder::create(
+            sprintf('/api/v2/%s/%s', $section, $resource),
+            HttpRequest::METHOD_GET,
+        );
+        $builder->withHeader('HTTP_ACCEPT', 'application/ld+json');
+
+        if ($token) {
+            $builder->withHeader('HTTP_' . $authorizationHeader, 'Bearer ' . $token);
+        }
+
+        return $builder->build();
+    }
+
+    public function subResourceIndex(string $section, string $resource, string $id, string $subResource): RequestInterface
+    {
+        $builder = RequestBuilder::create(
+            sprintf('/api/v2/%s/%s/%s/%s', $section, $resource, $id, $subResource),
+            HttpRequest::METHOD_GET,
+        );
+        $builder->withHeader('HTTP_ACCEPT', 'application/ld+json');
+
+        return $builder->build();
+    }
+
+    public function show(
+        string $section,
+        string $resource,
+        string $id,
+        string $authorizationHeader,
+        ?string $token = null
+    ): RequestInterface {
+        $builder = RequestBuilder::create(
+            sprintf('/api/v2/%s/%s/%s', $section, $resource, $id),
+            HttpRequest::METHOD_GET,
+        );
+        $builder->withHeader('HTTP_ACCEPT', 'application/ld+json');
+
+        if ($token) {
+            $builder->withHeader('HTTP_' . $authorizationHeader, 'Bearer ' . $token);
+        }
+
+        return $builder->build();
+    }
+
+    public function create(
+        string $section,
+        string $resource,
+        string $authorizationHeader,
+        ?string $token = null
+    ): RequestInterface {
+        $builder = RequestBuilder::create(
+            sprintf('/api/v2/%s/%s', $section, $resource),
+            HttpRequest::METHOD_POST,
+        );
+        $builder->withHeader('CONTENT_TYPE', 'application/ld+json');
```

Comment:
headers could go to const. probably protected if we want to leave this object not finalized